arctic/serialization/numpy_arrays.py

```diff
@@ -0,0 +1,179 @@
+import logging
+import numpy as np
+import numpy.ma as ma
+import pandas as pd
+
+from bson import Binary, SON
+
+from .._compress import compress, decompress
+
+
+class NumpyArrayConverter(object):
+    """
+    Converts a Numpy ndarray to and from PyMongo SON representation:
+
+        {
+         type:        '<i8',          # Numpy dtype (str)
+         length:      20,             # Numpy array length
+         values:      '\x00\x00bb...' # Compressed bytes
+        }
+    """
+
+    def docify(self, arr, meta=None):
+        """
+        Convert a Numpy ndarray to SON.
+
+        Parameters
+        ----------
+        arr:  ndarray
+            The numpy array to encode
+        meta: dict
+            Optional additional key value pairs to include in the SON
+        """
+        arr, mask = self._convert_types(arr)
+        son = SON(meta,
+                  type=arr.dtype.str,
+                  length=len(arr),
+                  values=Binary(compress(arr.tostring())))
+        if mask is not None:
+            son['mask'] = Binary(compress(mask.tostring()))
+        return son
+
+    def objify(self, doc):
+        """
+        Decode a Pymongo SON object into an Numpy ndarray
+        """
+        arr_data = decompress(doc['values'])
+        arr = np.fromstring(arr_data, doc['type'])
+
+        if 'mask' in doc:
```

Comment:
May be able to do a little better here by packing / unpacking bits:

https://github.com/manahl/arctic/blob/master/arctic/tickstore/tickstore.py#L627
https://github.com/manahl/arctic/blob/master/arctic/tickstore/tickstore.py#L427